stan/math/prim/scal/prob/normal_sufficient_lpdf.hpp

```diff
@@ -0,0 +1,175 @@
+                    "Scale parameter sufficient statistic", s_squared);
+      check_finite(function,
+                    "Number of observations", n_obs);
+      check_positive(function,
+                    "Number of observations", n_obs);
+      check_finite(function,
+                    "Location parameter", mu);
+      check_finite(function, "Scale parameter", sigma);
+      check_positive(function, "Scale parameter", sigma);
+      check_consistent_sizes(function,
+                             "Location parameter sufficient statistic",
+                             y_bar,
+                             "Scale parameter sufficient statistic",
+                             s_squared,
+                             "Number of observations", n_obs,
+                             "Location parameter", mu,
+                             "Scale parameter", sigma);
+      // check if no variables are involved and prop-to
+      if (!include_summand<propto, T_y, T_s, T_loc, T_scale>::value)
+        return 0.0;
+
+      // set up template expressions wrapping scalars into vector views
+      OperandsAndPartials<T_y, T_s, T_loc, T_scale>
+        operands_and_partials(y_bar, s_squared, mu, sigma);
+
+      scalar_seq_view<const T_y> y_bar_vec(y_bar);
+      scalar_seq_view<const T_s> s_squared_vec(s_squared);
+      scalar_seq_view<const T_n> n_obs_vec(n_obs);
+      scalar_seq_view<const T_loc> mu_vec(mu);
+      scalar_seq_view<const T_scale> sigma_vec(sigma);
+      size_t N = max_size(y_bar, s_squared, n_obs, mu, sigma);
+
+      for (size_t i = 0; i < N; i++) {
+        const T_partials_return y_bar_dbl = value_of(y_bar_vec[i]);
+        const T_partials_return s_squared_dbl =
+          value_of(s_squared_vec[i]);
+        const T_partials_return n_obs_dbl = n_obs_vec[i];
+        const T_partials_return mu_dbl = value_of(mu_vec[i]);
+        const T_partials_return sigma_dbl = value_of(sigma_vec[i]);
+        const T_partials_return sigma_squared = pow(sigma_dbl, 2);
+
+        if (include_summand<propto>::value)
+          logp += NEG_LOG_SQRT_TWO_PI * n_obs_dbl;
+
+        if (include_summand<propto, T_scale>::value)
+          logp -= n_obs_dbl*log(sigma_dbl);
+
+
+        const T_partials_return cons_expr =
+          (s_squared_dbl
+           + n_obs_dbl * pow(y_bar_dbl - mu_dbl, 2));
+
+        logp -= cons_expr / (2 * sigma_squared);
+
+
+        // gradients
+        if (!is_constant_struct<T_y>::value ||
+            !is_constant_struct<T_loc>::value) {
+          const T_partials_return common_derivative =
+            n_obs_dbl * (mu_dbl - y_bar_dbl) / sigma_squared;
+          if (!is_constant_struct<T_y>::value)
+            operands_and_partials.d_x1[i] += common_derivative;
+          if (!is_constant_struct<T_loc>::value)
+            operands_and_partials.d_x3[i] -= common_derivative;
+        }
+        if (!is_constant_struct<T_s>::value)
+          operands_and_partials.d_x2[i] -=
+            1 / (2 * sigma_squared);
```

Comment:
This would be more efficient as `0.5 / sigma_squared` --- it removes a multiplication.